### PR TITLE
Homework10: Alina Shautsova

### PR DESCRIPTION
**Pull request by AlinaShautsova**

### CODE REVIEW

**Review by Stml89 — CHANGES_REQUESTED** [review]
Please, merge ALL approved PRs

File: homework10_return_number.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+def result_of_function_is_number(func):
+    """Check if the function result is a number."""
+    def wrapper(*args, **kwargs):
+        """Display an error message, if function result is not a number"""
+        result = func(*args, **kwargs)
+        if not isinstance(result, (int, float)):
+            print(f"Error: This is {result} not a number")
+            return result
+        return result
```
Comment (Stml89) [inline homework10_return_number.py]:
```suggestion
        return result
```

File: homework10_type_decorator.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+def typed(type1):
+    """Function takes decorator arguments."""
+    def decorator(func):
+        """Function accepts the function to be decorated."""
+        def wrapper(*args):
+            """Function change type of arguments and execute main function."""
+            new_args = []
+            for arg in args:
+                new_args.append(type1(arg))
```
Comment (Stml89) [inline homework10_type_decorator.py]: JFYI: Function `map` can also help here

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework10_return_number.py (hunk 1 above)
Comment (AlinaShautsova) [reply homework10_return_number.py]: Done

File: homework10_type_decorator.py (hunk 2 above)
Comment (AlinaShautsova) [reply homework10_type_decorator.py]: Done

**Review by Stml89 — COMMENTED** [review]
approved

**Pull request merged**